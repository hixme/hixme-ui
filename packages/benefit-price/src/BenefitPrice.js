import React from 'react'
import PropTypes from 'prop-types'
import Price from '@hixme-ui/price'
import getLabel from './getLabel'
import getPrice from './getPrice'

const BenefitPrice = ({
  noLabel,
  payCycle,
  value,
  ...restProps
}) => (
  <Price
    value={getPrice(value, payCycle)}
    label={getLabel(payCycle)}
    {...restProps} 
  />
)
      
BenefitPrice.displayName = 'hui:BenefitPrice'
BenefitPrice.huiName = 'BenefitPrice'

BenefitPrice.propTypes = {
  value: PropTypes.number.isRequired,
  noLabel: PropTypes.bool,
  payCycle: PropTypes.number,
}

BenefitPrice.defaultProps = {
  noLabel: false,
  payCycle: 12,
}

export default BenefitPrice
