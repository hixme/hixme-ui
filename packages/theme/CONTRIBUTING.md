# Contributing

Open a PR at https://github.com/hixme/hixme-ui.
