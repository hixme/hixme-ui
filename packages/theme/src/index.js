export const colors = {
  white: 'white',
  black: 'black',
  grey30: '#3A4951',
  grey80: '#8F9DA8',
  grey100: '#AAB5BD',
  grey110: '#B9C5CD',
  grey120: '#CBD6DB',
  grey130: '#f7fcff',
  teal: '#2EAE8F',
  blue: '#3997E2',
  red: '#EB5565',
  yellow: '#ecb718',
  purple: '#A374D2',
  green: '#88b963',
}

const fontSizes = {
  jumbo: '22px',
  larger: '20px',
  large: '18px',
  medium: '16px',
  default: '14px',
  small: '12px',
  smaller: '10px',
  smallest: '8px',
}

const fontWeights = {
  default: 'normal',
  thinnest: '100',
  thinner: '200',
  thin: '300',
  normal: 'normal',
  bold: '700',
  bolder: '800',
  boldest: '900',
}

const titleFontSizes = {
  jumbo: '38px',
  larger: '36px',
  large: '34px',
  medium: '32px',
  default: '30px',
  small: '28px',
  smaller: '26px',
  smallest: '24px',
}

const buttonFontSizes = {
  mini: fontSizes.small,
  small: fontSizes.default,
  large: fontSizes.large,
  jumbo: fontSizes.jumbo,
}

const buttonHeights = {
  mini: '26px',
  small: '31px',
  default: '36px',
  large: '50px',
  jumbo: '65px',
}

const buttonMinWidths = {
  mini: '80px',
  small: '80px',
  default: '100px',
  large: '100px',
  jumbo: '100px',
}

const buttonTextColors = {
  default: colors.white,
}

const buttonFlatColors = {
  primary: colors.teal,
  secondary: colors.grey80,
  default: colors.teal,
  red: colors.red,
  danger: colors.red,
  error: colors.red,
  blue: colors.blue,
  gold: colors.yellow,
}

const shadows = {
  medium: '0px 2px 6px 0px rgba(56,59,60,0.22);',
  default: '0px 1px 4px 0px rgba(56,59,60,0.18);',
  none: '0px 0px 0px 0px rgba(0,0,0,0);',
}

const buttonBorderRadius = {
  none: '0',
  default: '2px',
}

const defaults = {
  fontSize: fontSizes.default,
  buttonHeight: buttonHeights.default,
  textColor: colors.grey30,
  fontWeight: fontWeights.default,
  backgroundColor: colors.white,
  borderColor: colors.grey80,
  titleFontSize: titleFontSizes.default,
}

export const colorTheme = {
  default: colors.grey30,
  primary: colors.teal,
  success: colors.teal,
  warning: colors.yellow,
  error: colors.red,
  danger: colors.red,
  muted: colors.grey80,

  white: colors.white,
  black: colors.black,
  red: colors.red,
  blue: colors.blue,
  purple: colors.purple,
  yellow: colors.yellow,
  green: colors.green,

  lighter: colors.grey110,
  light: colors.grey80,
  lightest: colors.grey130,
}

const outlineButtonColors = {
  ...colorTheme,
  default: colors.grey80,
}

const theme = {
  colors: colorTheme,
  backgroundColors: {
    ...colorTheme,
    default: defaults.backgroundColor,
  },
  borderColors: {
    ...colorTheme,
    default: defaults.borderColor,
  },
  buttonTextColors,
  buttonFlatColors,
  buttonFontSizes,
  buttonHeights,
  buttonMinWidths,
  buttonBorderRadius,
  outlineButtonColors,
  defaults,
  fontSizes,
  fontWeights,
  shadows,
  textColors: {
    ...colorTheme,
    default: defaults.textColor,
  },
  titleFontSizes,
}

export default theme

export ThemeProvider from './ThemeProvider'
