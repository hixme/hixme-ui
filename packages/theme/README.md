# hixme-ui theme

```bash
npm i --save @hixme-ui/theme
```

## Theme

HUI theme colors and defaults

```javascript
import theme from '@hixme-ui/theme

const primary = theme.colors.primary
```


## ThemeProvider

HUI theme provider, which is just an implementation of styled-components
ThemeProvider with the HUI theme as the default

```javascript
import { ThemeProvider } from '@hixme-ui/theme'

const App = ({ children }) => (
  <ThemeProvider>
    {children}
  </ThemeProvider>
)
```
