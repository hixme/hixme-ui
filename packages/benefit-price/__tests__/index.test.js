import React from 'react'
import renderer from 'react-test-renderer'
import { ThemeProvider } from '@hixme-ui/theme'
import BenefitPrice from '../src/index'
import 'jest-styled-components'

describe('BenefitPrice', () => {
  test('it renders', () => {
    const benefitPrice = renderer.create(<ThemeProvider><BenefitPrice value={100} /></ThemeProvider>)

    const tree = benefitPrice.toJSON()
    expect(tree.type).toBe('div')
  })

  test('it has 0px padding by default', () => {
    const component = renderer.create(
      <ThemeProvider><BenefitPrice value={4321} /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('padding', '0')
  })

  test('it should render with background: none', () => {
    const component = renderer.create(
      <ThemeProvider><BenefitPrice value={4321} /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('background', 'none')
  })

  test('it should render with display: flex', () => {
    const component = renderer.create(
      <ThemeProvider><BenefitPrice value={4321} /></ThemeProvider>,
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(tree).toHaveStyleRule('display', 'flex')
  })
})
