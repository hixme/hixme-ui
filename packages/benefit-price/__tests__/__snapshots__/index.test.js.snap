// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BenefitPrice it has 0px padding by default 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0;
  border: none;
}

.c1 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  margin: 5px 0;
  border: none;
}

.c2 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  border: none;
}

.c6 {
  background: none;
  padding: 0;
  margin: -10px 0 0 0;
  border: none;
}

.c7 {
  color: #8F9DA8;
  font-size: 8pt;
  font-weight: normal;
  border-bottom: none;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: inherit;
  text-align: inherit;
  text-indent: inherit;
  text-transform: none;
  text-overflow: inherit;
  -webkit-user-select: inherit;
  -moz-user-select: inherit;
  -ms-user-select: inherit;
  user-select: inherit;
  white-space: inherit;
  width: auto;
  word-break: inherit;
}

.c3 {
  font-size: 15pt;
  padding: 0 5px 0 0;
  font-weight: 300;
  color: #3A4951;
}

.c4 {
  font-size: 32pt;
  margin-top: -10px;
  color: #3A4951;
}

.c5 {
  font-size: 13pt;
  font-weight: bold;
  color: #3A4951;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
>
  <div
    className="c1"
    default={false}
    direction=""
    height={null}
    muted={false}
    width={null}
  >
    <div
      className="c2"
      default={false}
      direction=""
      height={null}
      muted={false}
      width={null}
    >
      <span
        className="c3"
        id="dollar-sign"
      >
        $
      </span>
      <span
        className="c4"
        id="dollars"
      >
        4321
      </span>
      <span
        className="c5"
        id="cents"
      >
        .
        00
      </span>
    </div>
    <div
      className="c6"
      default={false}
      direction=""
      height={null}
      muted={false}
      width={null}
    >
      <span
        className="c7"
        fontSize="8pt"
        href={null}
        id="label"
      >
        monthly
      </span>
    </div>
  </div>
</div>
`;

exports[`BenefitPrice it should render with background: none 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0;
  border: none;
}

.c1 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  margin: 5px 0;
  border: none;
}

.c2 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  border: none;
}

.c6 {
  background: none;
  padding: 0;
  margin: -10px 0 0 0;
  border: none;
}

.c7 {
  color: #8F9DA8;
  font-size: 8pt;
  font-weight: normal;
  border-bottom: none;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: inherit;
  text-align: inherit;
  text-indent: inherit;
  text-transform: none;
  text-overflow: inherit;
  -webkit-user-select: inherit;
  -moz-user-select: inherit;
  -ms-user-select: inherit;
  user-select: inherit;
  white-space: inherit;
  width: auto;
  word-break: inherit;
}

.c3 {
  font-size: 15pt;
  padding: 0 5px 0 0;
  font-weight: 300;
  color: #3A4951;
}

.c4 {
  font-size: 32pt;
  margin-top: -10px;
  color: #3A4951;
}

.c5 {
  font-size: 13pt;
  font-weight: bold;
  color: #3A4951;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
>
  <div
    className="c1"
    default={false}
    direction=""
    height={null}
    muted={false}
    width={null}
  >
    <div
      className="c2"
      default={false}
      direction=""
      height={null}
      muted={false}
      width={null}
    >
      <span
        className="c3"
        id="dollar-sign"
      >
        $
      </span>
      <span
        className="c4"
        id="dollars"
      >
        4321
      </span>
      <span
        className="c5"
        id="cents"
      >
        .
        00
      </span>
    </div>
    <div
      className="c6"
      default={false}
      direction=""
      height={null}
      muted={false}
      width={null}
    >
      <span
        className="c7"
        fontSize="8pt"
        href={null}
        id="label"
      >
        monthly
      </span>
    </div>
  </div>
</div>
`;

exports[`BenefitPrice it should render with display: flex 1`] = `
.c0 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  padding: 0;
  border: none;
}

.c1 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  margin: 5px 0;
  border: none;
}

.c2 {
  background: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0;
  border: none;
}

.c6 {
  background: none;
  padding: 0;
  margin: -10px 0 0 0;
  border: none;
}

.c7 {
  color: #8F9DA8;
  font-size: 8pt;
  font-weight: normal;
  border-bottom: none;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: inherit;
  text-align: inherit;
  text-indent: inherit;
  text-transform: none;
  text-overflow: inherit;
  -webkit-user-select: inherit;
  -moz-user-select: inherit;
  -ms-user-select: inherit;
  user-select: inherit;
  white-space: inherit;
  width: auto;
  word-break: inherit;
}

.c3 {
  font-size: 15pt;
  padding: 0 5px 0 0;
  font-weight: 300;
  color: #3A4951;
}

.c4 {
  font-size: 32pt;
  margin-top: -10px;
  color: #3A4951;
}

.c5 {
  font-size: 13pt;
  font-weight: bold;
  color: #3A4951;
}

<div
  className="c0"
  default={false}
  direction=""
  height={null}
  muted={false}
  width={null}
>
  <div
    className="c1"
    default={false}
    direction=""
    height={null}
    muted={false}
    width={null}
  >
    <div
      className="c2"
      default={false}
      direction=""
      height={null}
      muted={false}
      width={null}
    >
      <span
        className="c3"
        id="dollar-sign"
      >
        $
      </span>
      <span
        className="c4"
        id="dollars"
      >
        4321
      </span>
      <span
        className="c5"
        id="cents"
      >
        .
        00
      </span>
    </div>
    <div
      className="c6"
      default={false}
      direction=""
      height={null}
      muted={false}
      width={null}
    >
      <span
        className="c7"
        fontSize="8pt"
        href={null}
        id="label"
      >
        monthly
      </span>
    </div>
  </div>
</div>
`;
